[TASK] Align with new TYPO3 documentation standards

alexander-nitsche opened the pull request:
Relates: https://github.com/TYPO3-Documentation/T3DocTeam/issues/182

alexander-nitsche committed:
[TASK] Align with new TYPO3 documentation standards

- align README.md, Settings.cfg, Index.rst, Includes.txt
- align .editorconfig, .gitignore, composer.json, ext_emconf.php
- add Sitemap.rst, genindex.rst

See https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/GeneralConventions/FileStructure.html for further details.

alexander-nitsche committed:
[DOCS] Align images

alexander-nitsche committed:
[DOCS] Fix rendering warnings

alexander-nitsche committed:
[DOCS] Rename Includes.txt to Includes.rst.txt

alexander-nitsche committed:
[DOCS] Adjust some reST directives

alexander-nitsche commented:
What is missing: Insert composer package name "friendsoftypo3/widgets" into "Composer name of extension" field of the TER extension page configuration.

alexander-nitsche commented:
How it looks:
![Screen Shot 2022-04-05 at 11 07 36](https://user-images.githubusercontent.com/20297232/161719997-3b1612de-1ee3-46b2-93aa-dc8cb3247bed.png)

alexander-nitsche committed:
[TASK] Add TER + documentation + repository to composer.json

Add the common extension destinations to composer.json. Packagist
displays them in a prominent place.

alexander-nitsche commented:
Added new commit regarding additional sources (TER, repository and docs.typo3.org) in composer.json.

Relates: https://github.com/TYPO3-Documentation/T3DocTeam/issues/185

ohader commented on composer.json:
```diff
@@ -1,9 +1,14 @@
   "name": "friendsoftypo3/widgets",
   "type": "typo3-cms-extension",
-  "description": "Dashboard Widgets for TYPO3",
+  "description": "Widgets Collection",
```
I'd suggest to keep "Dashboard"... e.g. "Dashboard Widget Collection" or "Widget Collection for ext:dashboard"

alexander-nitsche replied:
The idea behind this change was to remove redundancy. In TYPO3 context widgets only appear on the dashboard - as far as i am aware of - so as long as this set, the keyword "dashboard" only adds redundancy here. See e.g. the "Matomo Widgets" project (https://extensions.typo3.org/extension/matomo_widgets/) which concluded the same.

ohader replied:
The extension you referenced says
> **Dashboard** widgets with Matomo reports

The term "widget" has used in combination with Fluid already and "widget" might exist in other areas as well as this term is very generic. Having an additional reference to "dashboard" just helps users to grasp the scope easily.

alexander-nitsche replied:
Yes, they use "dashboard" in their abstract as we do in the first sentence of the abstract, but not in the title.

But i did not encounter Fluid widgets, so it might be good to separate from them. Will add "dashboard" though.

alexander-nitsche replied:
Done.

ohader reviewed (changes requested):
Reflect "Dashboard" in package name

alexander-nitsche committed:
[TASK] Emphasize "dashboard widgets" in title

The keyword "dashboard" is reintroduced to clearly
distinguish these widgets from Fluid widgets.

ohader commented:
TER foo done
